packages/export/src/nodes/pivot.js

```diff
@@ -0,0 +1,176 @@
+import _ from 'lodash/fp.js'
+import F from 'futil'
+import ContextureClient from 'contexture-client'
+import {
+  getKey,
+  resultsForDrilldown,
+} from 'contexture-client/exampleTypes/pivot.js'
+import { andGroup, runWith, setFilterOnly } from '../utils.js'
+
+// TODO move to shared folder
+let getPageSize = (grouping) => {
+  let pageSizeGetters = {
+    fieldValues: _.getOr(10, 'size'),
+  }
+  return _.getOr(_.noop, grouping.type, pageSizeGetters)(grouping)
+}
+
+export let getGroupingSize = (
+  node,
+  groupingType,
+  cardinalityResult,
+  exportAllPages
+) => {
+  let getNested = _.get(groupingType)
+  let groupingCardinality = 1 // starting with 1 for the total column/row
+
+  F.walk(getNested)((groupResult, index, parents) => {
+    let groupCardinality = _.getOr(
+      _.size(getNested(groupResult)),
+      `${groupingType}GroupCount`,
+      groupResult
+    )
+    if (!groupCardinality) return
+
+    let grouping = _.get([groupingType, _.size(parents)], node)
+
+    if (!exportAllPages)
+      groupCardinality = _.min([groupCardinality, getPageSize(grouping)])
+
+    // TODO else approximate full cardinality by using the average size of the first page of each grouping
+
+    groupingCardinality = groupingCardinality + groupCardinality
+  })(cardinalityResult)
+
+  return groupingCardinality
+}
+
+export default async ({ service, tree, exportAllPages, ...node }) => {
+  let run = (node) => runWith(service, tree, node)
+
+  let getContextureClient = (tree, node) =>
+    ContextureClient(
+      {
+        service,
+        debounce: 0,
+      },
+      andGroup(setFilterOnly(tree), node)
+    )
+
+  let getGroupingResult = async (type) => {
+    let groups = _.cloneDeep(_.get(type, node))
+    let lastGroup = _.last(groups)
+    if (_.isEmpty(groups)) return {}
+
+    if (lastGroup.type === 'fieldValues')
```

Comment:
Do you still need this check? The elasticsearch provider handles other types now.

Reply:
We should keep it for now, because setting explicit `skip` for non `fieldValues` will skip the sub-aggregation and `groupCount` retrieval. Until we write the script aggregations for other types

Reply:
If we don't do that now, I'm not sure we ever will. Alternatively, we could support checking whether the type implements getGroupCount or not (which could even just live in the elasticsearch provider as better skip behavior - either as the only behavior for skip or a new type skip mode)